crates/nu-command/src/formats/from/delimited.rs

```diff
@@ -33,37 +39,51 @@ fn from_delimited_string_to_value(
         reader.headers()?.iter().map(String::from).collect()
     };
 
-    let mut rows = vec![];
-    for row in reader.records() {
-        let mut output_row = vec![];
-        for value in row?.iter() {
-            if no_infer {
-                output_row.push(Value::string(value.to_string(), span));
-                continue;
-            }
+    Ok(reader
+        .into_records()
+        .scan(
+            headers,
+            move |headers, row: csv::Result<csv::StringRecord>| {
+                let row = match row {
+                    Err(err) => {
+                        eprintln!("Error parsing CSV record: {}", err);
```

Comment:
Should we be using `ShellError` here?

Reply:
Yeah I tried this initially, but I wasn't sure how to handle the API. To stay streaming, this would have to become an `Iterator<Item = Result<Value>>`, which cannot be turned into a `PipelineData`.

I'm open to suggestions here, because I don't really like just swallowing the error.

Reply:
@dead10ck 
if you just want to print the error, like a warning of sort, you can use the `report_error_new` function :wink: